config/config.go

```diff
@@ -70,13 +71,45 @@ func IsMultipartMime(userdata string) bool {
 	return strings.Contains(contentType, "multipart/mixed")
 }
 
+// Normalize transform the "-" in "_" for the keys
+// in the YAML configuration.
+// e.g reboot-strategy -> reboot_strategy
+func Normalize(r io.Reader) string {
```

Comment:
Honestly, this is a rather risky function. Will it mess some inline bash script like:
```yaml
- some_inline_stuff: |
  echo 'foo-bar: stuff'
```

https://play.golang.com/p/u6ogWtIHI9Q

This kind of transform is probably done safe only by actually parsing YAML.

Reply:
I tend to agree here. If we mutate the source yaml, we're asking for trouble. It would be nice if the `yaml` package supported multiple tags for the same field, but sadly it does not.

Reply:
Personally, I'd drop this backward compatibility then, because continuing to support invalid keys adds too much to maintenance overhead. Otherwise we risk mangling unrelated stuff in config file. If you decide to drop it then we also will need to announce it. I think a changelog entry when bumping the commit ID in scripts repo would be sufficient.